### PR TITLE
Fix an issue with devices and dtypes checks not handling tuples properly

### LINKED ISSUE #20982: [Bug] Framework test suite failing to run because of supported devices and dtypes checks issue with tuple
Trying to run a framework test fails entirely because of an issue with functions that have the `with_unsupported_device_and_dtypes` decorator.

To reproduce the issue try to run `test_vector_norm` from `test_linalg`, the following is the error log:
```
/opt/miniconda/envs/multienv/bin/python3 /opt/.pycharm_helpers/pycharm/_jb_pytest_runner.py --target test_linalg.py::test_vector_norm -- -N 400 --jb-show-summary 
Testing started at 12:02 ...
/opt/.pycharm_helpers/pycharm/_jb_pytest_runner.py:8: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  from pkg_resources import iter_entry_points
Launching pytest with arguments -N 400 test_linalg.py::test_vector_norm in /opt/project/ivy_tests/test_ivy/test_functional/test_core

Use Database in ReadOnly Mode with local caching !
============================= test session starts ==============================
platform linux -- Python 3.10.0, pytest-7.4.0, pluggy-1.2.0 -- /opt/miniconda/envs/multienv/bin/python3
cachedir: .pytest_cache
hypothesis profile 'ivy_profile' -> print_blob=True, max_examples=400, deadline=timedelta(milliseconds=500000), suppress_health_check=[HealthCheck.too_slow, HealthCheck.filter_too_much, HealthCheck.data_too_large], database=MultiplexedDatabase(DirectoryBasedExampleDatabase('/opt/project/ivy_tests/test_ivy/test_functional/test_core/.hypothesis/examples/'), ReadOnlyDatabase(RedisExampleDatabase(Redis<ConnectionPool<Connection<host=redis-17011.c259.us-central1-2.gce.cloud.redislabs.com,port=17011,db=0>>>, expire_after=datetime.timedelta(days=8))))
metadata: {'Python': '3.10.0', 'Platform': 'Linux-5.15.49-linuxkit-pr-x86_64-with-glibc2.28', 'Packages': {'pytest': '7.4.0', 'pluggy': '1.2.0'}, 'Plugins': {'anyio': '3.7.1', 'hypothesis': '6.82.0', 'json-report': '1.5.0', 'metadata': '3.0.0'}}
backend(s): all
device: cpu
number of Hypothesis examples: 400
rootdir: /opt/project
plugins: anyio-3.7.1, hypothesis-6.82.0, json-report-1.5.0, metadata-3.0.0
collecting ... collected 0 items / 1 error

==================================== ERRORS ====================================
_ ERROR collecting ivy_tests/test_ivy/test_functional/test_core/test_linalg.py _
../../../../ivy/utils/exceptions.py:335: in _handle_exceptions
    return fn(*args, **kwargs)
../../../../ivy/func_wrapper.py:976: in _handle_nestable
    return fn(*args, **kwargs)
../../../../ivy/functional/ivy/general.py:4030: in function_supported_devices_and_dtypes
    _is_valid_device_and_dtypes_attributes(fn),
../../../../ivy/functional/ivy/general.py:3886: in _is_valid_device_and_dtypes_attributes
    if isinstance(list(fn_unsupported_dnd.__get__().values())[0], dict):
E   AttributeError: 'tuple' object has no attribute 'values'

During handling of the above exception, another exception occurred:
../../../../ivy/utils/exceptions.py:335: in _handle_exceptions
    return fn(*args, **kwargs)
../../../../ivy/func_wrapper.py:976: in _handle_nestable
    return fn(*args, **kwargs)
../../../../ivy/functional/ivy/general.py:4047: in function_supported_devices_and_dtypes
    supported_devices_dtypes = ivy.functional.data_type._nested_get(
../../../../ivy/functional/ivy/data_type.py:151: in _nested_get
    f_supported = get_fn(fn, False)
../../../../ivy/utils/exceptions.py:362: in _handle_exceptions
    raise ivy.utils.exceptions.IvyAttributeError(
E   ivy.utils.exceptions.IvyAttributeError: paddle: function_supported_devices_and_dtypes: 'tuple' object has no attribute 'values'

During handling of the above exception, another exception occurred:
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/runner.py:341: in from_call
    result: Optional[TResult] = func()
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/runner.py:372: in <lambda>
    call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/python.py:531: in collect
    self._inject_setup_module_fixture()
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/python.py:545: in _inject_setup_module_fixture
    self.obj, ("setUpModule", "setup_module")
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/python.py:310: in obj
    self._obj = obj = self._getobj()
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/python.py:528: in _getobj
    return self._importtestmodule()
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/python.py:617: in _importtestmodule
    mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
/opt/miniconda/envs/multienv/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
test_linalg.py:294: in <module>
    def test_matrix_power(*, dtype_x, n, test_flags, backend_fw, fn_name, on_device):
../../helpers/testing_helpers.py:364: in test_wrapper
    supported_device_dtypes = _get_supported_devices_dtypes(fn_name, fn_mod)
../../helpers/testing_helpers.py:234: in _get_supported_devices_dtypes
    devices_and_dtypes = ivy_backend.function_supported_devices_and_dtypes(_fn)
../../../../ivy/utils/exceptions.py:362: in _handle_exceptions
    raise ivy.utils.exceptions.IvyAttributeError(
E   ivy.utils.exceptions.IvyAttributeError: paddle: function_supported_devices_and_dtypes: paddle: function_supported_devices_and_dtypes: 'tuple' object has no attribute 'values'
=========================== short test summary info ============================
ERROR test_linalg.py - ivy.utils.exceptions.IvyAttributeError: paddle: functi...
============================== 1 error in 18.58s ===============================

ivy_tests/test_ivy/test_functional/test_core/test_linalg.py:None (ivy_tests/test_ivy/test_functional/test_core/test_linalg.py)
../../../../ivy/utils/exceptions.py:335: in _handle_exceptions
    return fn(*args, **kwargs)
../../../../ivy/func_wrapper.py:976: in _handle_nestable
    return fn(*args, **kwargs)
../../../../ivy/functional/ivy/general.py:4030: in function_supported_devices_and_dtypes
    _is_valid_device_and_dtypes_attributes(fn),
../../../../ivy/functional/ivy/general.py:3886: in _is_valid_device_and_dtypes_attributes
    if isinstance(list(fn_unsupported_dnd.__get__().values())[0], dict):
E   AttributeError: 'tuple' object has no attribute 'values'

During handling of the above exception, another exception occurred:
../../../../ivy/utils/exceptions.py:335: in _handle_exceptions
    return fn(*args, **kwargs)
../../../../ivy/func_wrapper.py:976: in _handle_nestable
    return fn(*args, **kwargs)
../../../../ivy/functional/ivy/general.py:4047: in function_supported_devices_and_dtypes
    supported_devices_dtypes = ivy.functional.data_type._nested_get(
../../../../ivy/functional/ivy/data_type.py:151: in _nested_get
    f_supported = get_fn(fn, False)
../../../../ivy/utils/exceptions.py:362: in _handle_exceptions
    raise ivy.utils.exceptions.IvyAttributeError(
E   ivy.utils.exceptions.IvyAttributeError: paddle: function_supported_devices_and_dtypes: 'tuple' object has no attribute 'values'

During handling of the above exception, another exception occurred:
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/runner.py:341: in from_call
    result: Optional[TResult] = func()
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/runner.py:372: in <lambda>
    call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/python.py:531: in collect
    self._inject_setup_module_fixture()
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/python.py:545: in _inject_setup_module_fixture
    self.obj, ("setUpModule", "setup_module")
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/python.py:310: in obj
    self._obj = obj = self._getobj()
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/python.py:528: in _getobj
    return self._importtestmodule()
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/python.py:617: in _importtestmodule
    mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
/opt/miniconda/envs/multienv/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
test_linalg.py:294: in <module>
    def test_matrix_power(*, dtype_x, n, test_flags, backend_fw, fn_name, on_device):
../../helpers/testing_helpers.py:364: in test_wrapper
    supported_device_dtypes = _get_supported_devices_dtypes(fn_name, fn_mod)
../../helpers/testing_helpers.py:234: in _get_supported_devices_dtypes
    devices_and_dtypes = ivy_backend.function_supported_devices_and_dtypes(_fn)
../../../../ivy/utils/exceptions.py:362: in _handle_exceptions
    raise ivy.utils.exceptions.IvyAttributeError(
E   ivy.utils.exceptions.IvyAttributeError: paddle: function_supported_devices_and_dtypes: paddle: function_supported_devices_and_dtypes: 'tuple' object has no attribute 'values'
ERROR: not found: /opt/project/ivy_tests/test_ivy/test_functional/test_core/test_linalg.py::test_vector_norm
(no name '/opt/project/ivy_tests/test_ivy/test_functional/test_core/test_linalg.py::test_vector_norm' in any of [<Module test_linalg.py>])


Process finished with exit code 4
```

### PR DESCRIPTION
**Pull request by fabriziocacicia**
Fix #20982

### CODE REVIEW

**Comment by 4di03** [comment]
Hi, I am facing the same issue. After adding these changes, it seems that any test ran with jax and paddle backend are failing with the following error message:
```
hypothesis.errors.InvalidArgument: Cannot sample from a length-zero sequence.
```

**Comment by fabriziocacicia** [comment]
> Hi, I am facing the same issue. After adding these changes, it seems that any test ran with jax and paddle backend are failing with the following error message:
> 
> ```
> hypothesis.errors.InvalidArgument: Cannot sample from a length-zero sequence.
> ```

@4di03 can you attach the entire log of the test failure? Thanks

**Comment by 4di03** [comment]
> > Hi, I am facing the same issue. After adding these changes, it seems that any test ran with jax and paddle backend are failing with the following error message:
> > ```
> > hypothesis.errors.InvalidArgument: Cannot sample from a length-zero sequence.
> > ```
> 
> @4di03 can you attach the entire log of the test failure? Thanks

Sure! 

```
/Users/adithyapalle/anaconda3/envs/ivy_dev/bin/python /Applications/PyCharm.app/Contents/plugins/python/helpers/pycharm/_jb_pytest_runner.py --target ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py::test_torch_bmm 
Testing started at 3:38 PM ...
/Applications/PyCharm.app/Contents/plugins/python/helpers/pycharm/_jb_pytest_runner.py:8: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  from pkg_resources import iter_entry_points
Launching pytest with arguments ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py::test_torch_bmm --no-header --no-summary -q in /Users/adithyapalle/Desktop/work/ivy

Use Database in ReadOnly Mode with local caching !
============================= test session starts ==============================
collecting ... collected 5 items

ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py::test_torch_bmm[cpu-numpy-False-False] 
ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py::test_torch_bmm[cpu-jax-False-False] 
ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py::test_torch_bmm[cpu-tensorflow-False-False] 
ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py::test_torch_bmm[cpu-torch-False-False] 
ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py::test_torch_bmm[cpu-paddle-False-False] 

========================= 2 failed, 3 passed in 8.43s ==========================
PASSED [ 20%]FAILED [ 40%]
ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py:395 (test_torch_bmm[cpu-jax-False-False])
on_device = 'cpu', frontend = 'torch', backend_fw = 'jax'

    @handle_frontend_test(
>       fn_tree="torch.bmm",
        dtype_and_matrices=_get_dtype_and_3dbatch_matrices(),
    )

ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py:397: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py:90: in _get_dtype_and_3dbatch_matrices
    dim_size1 = draw(helpers.ints(min_value=2, max_value=5))
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:951: in draw
    return strategy.do_draw(self)
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/strategies/_internal/lazy.py:160: in do_draw
    return data.draw(self.wrapped_strategy)
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:951: in draw
    return strategy.do_draw(self)
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/strategies/_internal/core.py:1539: in do_draw
    return self.definition(data.draw, *self.args, **self.kwargs)
ivy_tests/test_ivy/helpers/hypothesis_helpers/number_helpers.py:194: in ints
    dtype = draw(
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:951: in draw
    return strategy.do_draw(self)
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/strategies/_internal/lazy.py:160: in do_draw
    return data.draw(self.wrapped_strategy)
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:951: in draw
    return strategy.do_draw(self)
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/strategies/_internal/core.py:1539: in do_draw
    return self.definition(data.draw, *self.args, **self.kwargs)
ivy_tests/test_ivy/helpers/hypothesis_helpers/dtype_helpers.py:238: in get_dtypes
    return [draw(st.sampled_from(valid_dtypes[index:]))]
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:937: in draw
    strategy.validate()
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/strategies/_internal/strategies.py:417: in validate
    self.do_validate()
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/strategies/_internal/lazy.py:133: in do_validate
    w = self.wrapped_strategy
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/strategies/_internal/lazy.py:108: in wrapped_strategy
    base = self.function(*self.__args, **self.__kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

elements = []

    @defines_strategy(try_non_lazy=True)  # noqa: F811
    def sampled_from(
        elements: Union[Type[enum.Enum], Sequence[Any]]
    ) -> SearchStrategy[Any]:
        """Returns a strategy which generates any value present in ``elements``.
    
        Note that as with :func:`~hypothesis.strategies.just`, values will not be
        copied and thus you should be careful of using mutable data.
    
        ``sampled_from`` supports ordered collections, as well as
        :class:`~python:enum.Enum` objects.  :class:`~python:enum.Flag` objects
        may also generate any combination of their members.
    
        Examples from this strategy shrink by replacing them with values earlier in
        the list. So e.g. ``sampled_from([10, 1])`` will shrink by trying to replace
        1 values with 10, and ``sampled_from([1, 10])`` will shrink by trying to
        replace 10 values with 1.
    
        It is an error to sample from an empty sequence, because returning :func:`nothing`
        makes it too easy to silently drop parts of compound strategies.  If you need
        that behaviour, use ``sampled_from(seq) if seq else nothing()``.
        """
        values = check_sample(elements, "sampled_from")
        if not values:
            if (
                isinstance(elements, type)
                and issubclass(elements, enum.Enum)
                and vars(elements).get("__annotations__")
            ):
                # See https://github.com/HypothesisWorks/hypothesis/issues/2923
                raise InvalidArgument(
                    f"Cannot sample from {elements.__module__}.{elements.__name__} "
                    "because it contains no elements.  It does however have annotations, "
                    "so maybe you tried to write an enum as if it was a dataclass?"
                )
>           raise InvalidArgument("Cannot sample from a length-zero sequence.")
E           hypothesis.errors.InvalidArgument: Cannot sample from a length-zero sequence.
E           
E           You can reproduce this example by temporarily adding @reproduce_failure('6.82.0', b'AA==') as a decorator on your test case

../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/strategies/_internal/core.py:204: InvalidArgument
PASSED [ 60%]PASSED [ 80%]FAILED [100%]
ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py:395 (test_torch_bmm[cpu-paddle-False-False])
on_device = 'cpu', frontend = 'torch', backend_fw = 'paddle'

    @handle_frontend_test(
>       fn_tree="torch.bmm",
        dtype_and_matrices=_get_dtype_and_3dbatch_matrices(),
    )

ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py:397: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ivy_tests/test_ivy/test_frontends/test_torch/test_blas_and_lapack_ops.py:90: in _get_dtype_and_3dbatch_matrices
    dim_size1 = draw(helpers.ints(min_value=2, max_value=5))
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:951: in draw
    return strategy.do_draw(self)
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/strategies/_internal/lazy.py:160: in do_draw
    return data.draw(self.wrapped_strategy)
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:951: in draw
    return strategy.do_draw(self)
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/strategies/_internal/core.py:1539: in do_draw
    return self.definition(data.draw, *self.args, **self.kwargs)
ivy_tests/test_ivy/helpers/hypothesis_helpers/number_helpers.py:194: in ints
    dtype = draw(
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:951: in draw
    return strategy.do_draw(self)
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/strategies/_internal/lazy.py:160: in do_draw
    return data.draw(self.wrapped_strategy)
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:951: in draw
    return strategy.do_draw(self)
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/strategies/_internal/core.py:1539: in do_draw
    return self.definition(data.draw, *self.args, **self.kwargs)
ivy_tests/test_ivy/helpers/hypothesis_helpers/dtype_helpers.py:238: in get_dtypes
    return [draw(st.sampled_from(valid_dtypes[index:]))]
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/internal/conjecture/data.py:937: in draw
    strategy.validate()
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/strategies/_internal/strategies.py:417: in validate
    self.do_validate()
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/strategies/_internal/lazy.py:133: in do_validate
    w = self.wrapped_strategy
../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/strategies/_internal/lazy.py:108: in wrapped_strategy
    base = self.function(*self.__args, **self.__kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

elements = []

    @defines_strategy(try_non_lazy=True)  # noqa: F811
    def sampled_from(
        elements: Union[Type[enum.Enum], Sequence[Any]]
    ) -> SearchStrategy[Any]:
        """Returns a strategy which generates any value present in ``elements``.
    
        Note that as with :func:`~hypothesis.strategies.just`, values will not be
        copied and thus you should be careful of using mutable data.
    
        ``sampled_from`` supports ordered collections, as well as
        :class:`~python:enum.Enum` objects.  :class:`~python:enum.Flag` objects
        may also generate any combination of their members.
    
        Examples from this strategy shrink by replacing them with values earlier in
        the list. So e.g. ``sampled_from([10, 1])`` will shrink by trying to replace
        1 values with 10, and ``sampled_from([1, 10])`` will shrink by trying to
        replace 10 values with 1.
    
        It is an error to sample from an empty sequence, because returning :func:`nothing`
        makes it too easy to silently drop parts of compound strategies.  If you need
        that behaviour, use ``sampled_from(seq) if seq else nothing()``.
        """
        values = check_sample(elements, "sampled_from")
        if not values:
            if (
                isinstance(elements, type)
                and issubclass(elements, enum.Enum)
                and vars(elements).get("__annotations__")
            ):
                # See https://github.com/HypothesisWorks/hypothesis/issues/2923
                raise InvalidArgument(
                    f"Cannot sample from {elements.__module__}.{elements.__name__} "
                    "because it contains no elements.  It does however have annotations, "
                    "so maybe you tried to write an enum as if it was a dataclass?"
                )
>           raise InvalidArgument("Cannot sample from a length-zero sequence.")
E           hypothesis.errors.InvalidArgument: Cannot sample from a length-zero sequence.
E           
E           You can reproduce this example by temporarily adding @reproduce_failure('6.82.0', b'AA==') as a decorator on your test case

../../../anaconda3/envs/ivy_dev/lib/python3.10/site-packages/hypothesis/strategies/_internal/core.py:204: InvalidArgument

Process finished with exit code 1
```

File: ivy/functional/ivy/general.py
Comment (4di03) [inline ivy/functional/ivy/general.py:1]: I think Line 3893 (fn_supported_dnd.__get__()) also needs this check.

**Review by vedpatwardhan — COMMENTED** [review]
I tried running `test_vector_norm` with the current version on `master` and it seems to be passing, is the root cause still persistent? Thanks @fabriziocacicia @4di03 😄

**Comment by Tonycrux** [comment]
Hi I am having same issue while running test in the jax framework. 

Is the issue fixed now?

Here is the log

```
____________________________________________ ERROR collecting ivy_tests/test_ivy/test_frontends/test_jax/test_lax/test_operators.py ____________________________________________
../../../../../ivy/utils/exceptions.py:335: in _handle_exceptions
    return fn(*args, **kwargs)
../../../../../ivy/func_wrapper.py:976: in _handle_nestable
    return fn(*args, **kwargs)
../../../../../ivy/functional/ivy/general.py:4030: in function_supported_devices_and_dtypes
    _is_valid_device_and_dtypes_attributes(fn),
../../../../../ivy/functional/ivy/general.py:3891: in _is_valid_device_and_dtypes_attributes
    if isinstance(list(fn_supported_dnd.__get__().values())[0], dict):
E   AttributeError: 'tuple' object has no attribute 'values'

During handling of the above exception, another exception occurred:
../../../../../ivy/utils/exceptions.py:335: in _handle_exceptions
    return fn(*args, **kwargs)
../../../../../ivy/func_wrapper.py:976: in _handle_nestable
    return fn(*args, **kwargs)
../../../../../ivy/functional/ivy/general.py:4047: in function_supported_devices_and_dtypes
    supported_devices_dtypes = ivy.functional.data_type._nested_get(
../../../../../ivy/functional/ivy/data_type.py:151: in _nested_get
    f_supported = get_fn(fn, False)
../../../../../ivy/utils/exceptions.py:362: in _handle_exceptions
    raise ivy.utils.exceptions.IvyAttributeError(
E   ivy.utils.exceptions.IvyAttributeError: paddle: function_supported_devices_and_dtypes: 'tuple' object has no attribute 'values'

During handling of the above exception, another exception occurred:
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/runner.py:341: in from_call
    result: Optional[TResult] = func()
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/runner.py:372: in <lambda>
    call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/python.py:531: in collect
    self._inject_setup_module_fixture()
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/python.py:545: in _inject_setup_module_fixture
    self.obj, ("setUpModule", "setup_module")
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/python.py:310: in obj
    self._obj = obj = self._getobj()
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/python.py:528: in _getobj
    return self._importtestmodule()
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/python.py:617: in _importtestmodule
    mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/pathlib.py:565: in import_path
    importlib.import_module(module_name)
/opt/miniconda/envs/multienv/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
test_operators.py:14: in <module>
    from ivy_tests.test_ivy.test_functional.test_experimental.test_nn.test_layers import (
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
/opt/miniconda/envs/multienv/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
../../../test_functional/test_experimental/test_nn/test_layers.py:91: in <module>
    def test_max_pool2d(
../../../helpers/testing_helpers.py:364: in test_wrapper
    supported_device_dtypes = _get_supported_devices_dtypes(fn_name, fn_mod)
../../../helpers/testing_helpers.py:234: in _get_supported_devices_dtypes
    devices_and_dtypes = ivy_backend.function_supported_devices_and_dtypes(_fn)
../../../../../ivy/utils/exceptions.py:362: in _handle_exceptions
    raise ivy.utils.exceptions.IvyAttributeError(
E   ivy.utils.exceptions.IvyAttributeError: paddle: function_supported_devices_and_dtypes: paddle: function_supported_devices_and_dtypes: 'tuple' object has no attribute 'values'
=========================================================================== short test summary info ============================================================================
ERROR test_operators.py - ivy.utils.exceptions.IvyAttributeError: paddle: function_supported_devices_and_dtypes: paddle: function_supported_devices_and_dtypes: 'tuple' object has no attribute 'values'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

**Comment by 4di03** [comment]
> I tried running `test_vector_norm` with the current version on `master` and it seems to be passing, is the root cause still persistent? Thanks @fabriziocacicia @4di03 😄

Yes, it seems that it is working on the latest version of master. However, it seems that Jax backend is still failing with either KeyError: 'cpu' or the other error message I shared above, but that is probably a separate issue. Thanks!

**Comment by fabriziocacicia** [comment]
> > I tried running `test_vector_norm` with the current version on `master` and it seems to be passing, is the root cause still persistent? Thanks @fabriziocacicia @4di03 😄
> 
> 
> 
> It seems that the issue is resolved on latest version of master. However, it seems that Jax backend is still failing with either KeyError: 'cpu' or the other error message I shared above, but that is probably a separate issue. Thanks!

Same for me. I think I will close this.

**Pull request closed without merging**